doc/sphinx/source/conformance.rst

```diff
@@ -0,0 +1,186 @@
+
+Known issues in pocl / things to be aware of
+--------------------------------------------
+
+- Integer division by zero. OpenCL 1.2 specification requires that division by
+  zero on integers results in undefined values, instead of raising exceptions.
+  This requires pocl to install a handler of SIGFPE. Unfortunately signal
+  handlers are per-process not per-thread, and pocl drivers do not run in a
+  separate process, which means that integer division by zero will not raise
+  SIGFPE for the entire pocl library and also the user's program. The handler
+  may be disabled by setting the env variable POCL_SIGFPE_HANDLER to 0.
+  Note that this is currently only relevant for x86(-64) + Linux, on all other
+  systems this issue is not handled in any way (thus Pocl is likely
+  non-conformant there).
+
+- Several options to clBuildProgram() are accepted but currently have no effect.
+  This is related mostly to optimization options like `-cl-fast-relaxed-math`.
+  The `-cl-denorms-are-zero` and `-cl-fp32-correctly-rounded-divide-sqrt`
+  options are honored.
+
+- Many of ``native_`` and ``half_`` variants of kernel library functions are mapped
+  to the "full" variants.
+
+- the optional OpenGL interop extension is not supported
+
+Conformance tests results (kernel library precision) on tested hardware
+-----------------------------------------------------------------------
+
+Note that it's impossible to test double precision on the entire range,
+therefore the results may vary.
+
+x86-64 CPU with AVX2+FMA
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+====================   =========================   ===========================================================
+        NAME                Worst ULP                 WHERE
+====================   =========================   ===========================================================
+             add            0.00                      {0x0p+0, 0x0p+0}
+             addD           0.00                      {0x0p+0, 0x0p+0}
+      assignment            0.00                      0x0p+0
+      assignmentD           0.00                      0x0p+0
+            cbrt            0.50                      -0x1.5629d2p+116
+            cbrtD           0.59                      0x1.0000000000136p+1022
+            ceil            0.00                      0x0p+0
+            ceilD           0.00                      0x0p+0
+        copysign            0.00                      {0x0p+0, 0x0p+0}
+        copysignD           0.00                      {0x0p+0, 0x0p+0}
+             cos            2.37                      0x1.1338ccp+20
+             cosD           2.27                      -0x1.d10000000074p+380
+            cosh            2.41                      -0x1.602166p+2
+            coshD           1.43                      -0x1.98000000003efp+5
+           cospi            1.94                      0x1.d73b56p-2
+           cospiD           2.46                      -0x1.adffffffffa91p-2
```

Comment:
Are these errors still within the allowed limits?

Reply:
yes, cosh & cospi are <= 4 ULP. this is formatted output from the math conformance test